### PR TITLE
Added [ method to xcmsSet and fixed a problem in split.xcmsSet

### PR DESCRIPTION
**Pull request by jotsetung**
+ I've implemented a [ method that allows to subset an xcmsSet.
+ Fixed a problem in split.xcmsSet that did not split the phenoData
properly.
+ Added some details to the documentation of xcmsSet-class.

hope you find the changes useful.
cheers, jo

### CODE REVIEW

File: R/xcmsSet.R [hunk 1]
```diff
@@ -253,6 +253,18 @@ split.xcmsSet <- function(x, f, drop = TRUE, ...) {
     lcsets <- vector("list", length(levels(f)))
     names(lcsets) <- levels(f)
 
+    ## get the phenoData and all other parameters we want to pass
+    ## down to the splitted objects
+    pd <- phenoData(x)
+    dataCor <- x@dataCorrection
+    suppressWarnings(
```
Comment (sneumann) [inline R/xcmsSet.R]: Which warnings are expected here ? Please add to the comment, same three lines below.

File: R/xcmsSet.R [hunk 2]
```diff
@@ -2296,3 +2323,157 @@ setMethod("profMethod", "xcmsSet", function(object) {
...
+  ## 5) filled
+  if(length(x@filled) > 0){
+      xsub@filled <- (1:nrow(keep.peaks))[rownames(keep.peaks) %in% as.character(x@filled)]
+  }
+  ## 6) dataCorrection
+  if(length(x@dataCorrection)>0)
+      xset@dataCorrection <- x@dataCorrection[j]
+  return(xsub)
+})
+## setMethod("[", "eSet", function(x, i, j, ..., drop = FALSE) {
```
Comment (sneumann) [inline R/xcmsSet.R]: Once confident with the above, just remove this block in several weeks

File: man/xcmsSet-class.Rd [hunk 3]
```diff
@@ -231,11 +239,21 @@
...
+      \code{signature("xcmsSet")}: divide the xcmsSet into a list of
+      xcmsSet objects depending on the provided factor. Note that only
+      peak data will be preserved, i.e. eventual peak grouping information
+      will be lost.
     }
     \item{\code{object$name}, \code{object$name<-value}}{
       Access and set \code{name} column in \code{phenoData}
     }
+    \item{\code{object[, i]}}{
+      Conducts subsetting of a \code{xcmsSet} instance. Only subsetting
```
Comment (sneumann) [inline man/xcmsSet-class.Rd]: After subsetting the minfrac might be violated, right ?

File: R/xcmsSet.R (hunk 1 above)
Comment (jotsetung) [inline R/xcmsSet.R]: I've added the mslevel and scanrange slots in a previous commit, thus, they are not present in "older" xcmsSet objects (like the ones from the faahKO package). When the methods mslevel and scanrange are called on such objects a warning is generated stating that the slots are not available and returning the standard values for these slots.

File: R/xcmsSet.R (hunk 2 above)
Comment (jotsetung) [inline R/xcmsSet.R]: a yes, right. The commented out code can be removed straight away... sorry, forgot about that.

File: man/xcmsSet-class.Rd (hunk 3 above)
Comment (jotsetung) [inline man/xcmsSet-class.Rd]: yes, the subsetting of the groups might be problematic after all.

**Pull request merged**